core/src/main/java/org/axonframework/messaging/annotation/AnnotatedHandlerInspector.java

```diff
@@ -57,6 +71,22 @@ private AnnotatedHandlerInspector(Class<T> inspectedType,
         return inspectType(handlerType, ClasspathParameterResolverFactory.forClass(handlerType));
     }
 
+    /**
+     * Create an inspector for given {@code handlerType} that uses a {@link ClasspathParameterResolverFactory} to
+     * resolve method parameters and given {@code handlerDefinition} to create handlers.
+     *
+     * @param handlerType       the target handler type
+     * @param handlerDefinition the handler definition used to create concrete handlers
+     * @param <T>               the handler's type
+     * @return a new inspector instance for the inspected class
+     */
+    public static <T> AnnotatedHandlerInspector<T> inspectType(Class<? extends T> handlerType,
```

Comment:
Don't think this constructor is necessary. Anyone passing in a `HandlerDefinition` may also be expected to pass in a `ParameterResolverFactory`.